src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -8544,7 +8544,9 @@ extern "C" bool QCALLTYPE SfiNext(StackFrameIterator* pThis, uint* uExCollideCla
             _ASSERTE(retVal != SWA_FAILED);
             _ASSERTE(pThis->GetFrameState() != StackFrameIterator::SFITER_SKIPPED_FRAME_FUNCTION);
 
-            if (pThis->m_crawl.GetFrame() == FRAME_TOP)
+            pFrame = pThis->m_crawl.GetFrame();
+            // Check if there are any further managed frames on the stack, if not, the exception is unhandled.
+            if ((pFrame == FRAME_TOP) || (pFrame->GetVTablePtr() == DebuggerU2MCatchHandlerFrame::GetMethodFrameVPtr()))
```

Comment:
Can `DebuggerU2MCatchHandlerFrame` be in the middle of the stack, with a bunch more managed frame behind i?

Reply:
I have believed when I've searched some time ago that the only place where it is used is on top of secondary threads. But I have searched for it again to be 100% sure now and I have actually missed on in `DispatchInfo::InvokeMember`. So it seems that both this check and a similar one in the `NotifyExceptionPassStarted` should also check if that frame is also the topmost one.

Reply:
@jkotas I have added check for the DebuggerU2MCatchHandlerFrame being the topmost one.